core/jobs.py

```diff
@@ -1059,6 +1307,7 @@ def on_batch_job_canceled(cls):
 
     @classmethod
     def on_batch_job_failure(cls):
+        """Gives up on job and delegates responsibility to previous job."""
```

Comment:
No, that's not what this does...

Gives up on the existing batch job, and kicks off a new one.